pydis_site/apps/content/utils.py

```diff
@@ -27,8 +55,258 @@ def get_categories(path: Path) -> Dict[str, Dict]:
+        else:
+            tags = fetch_tags()
+            record_tags(tags)
+
+        return tags
+    else:
+        # Get tags from database
+        return Tag.objects.all()
+
+
+def get_tag(path: str, *, skip_sync: bool = False) -> typing.Union[Tag, list[Tag]]:
+    """
+    Return a tag based on the search location.
+
+    If certain tag data is out of sync (for instance a commit date is missing),
+    an extra request will be made to sync the information.
+
+    The tag name and group must match. If only one argument is provided in the path,
+    it's assumed to either be a group name, or a no-group tag name.
+
+    If it's a group name, a list of tags which belong to it is returned.
+    """
+    path = path.split("/")
+    if len(path) == 2:
+        group, name = path[0], path[1]
+    else:
+        name = path[0]
+        group = None
+
+    matches = []
+    for tag in get_tags():
+        if tag.name == name and tag.group == group:
+            if tag.last_commit is None and not skip_sync:
+                set_tag_commit(tag)
+            return tag
+        elif tag.group == name and group is None:
+            matches.append(tag)
+
+    if matches:
+        return matches
+
+    raise Tag.DoesNotExist()
+
+
+def get_tag_category(
+    tags: typing.Optional[list[Tag]] = None, *, collapse_groups: bool
+) -> dict[str, dict]:
+    """
+    Generate context data for `tags`, or all tags if None.
+
+    If `tags` is None, `get_tag` is used to populate the data.
+    If `collapse_groups` is True, tags with parent groups are not included in the list,
+    and instead the parent itself is included as a single entry with it's sub-tags
+    in the description.
+    """
+    if not tags:
+        tags = get_tags()
+
+    data = []
+    groups = {}
+
+    # Create all the metadata for the tags
+    for tag in tags:
+        if tag.group is None or not collapse_groups:
+            content = frontmatter.parse(tag.body)[1]
+            data.append({
+                "title": tag.name,
+                "description": markdown.markdown(content, extensions=["pymdownx.superfences"]),
+                "icon": "fas fa-tag",
+            })
+        else:
+            if tag.group not in groups:
+                groups[tag.group] = {
+                    "title": tag.group,
+                    "description": [tag.name],
+                    "icon": "fas fa-tags",
+                }
+            else:
+                groups[tag.group]["description"].append(tag.name)
+
+    # Flatten group description into a single string
+    for group in groups.values():
+        # If the following string is updated, make sure to update it in the frontend JS as well
+        group["description"] = "Contains the following tags: " + ", ".join(group["description"])
+        data.append(group)
+
+    # Sort the tags, and return them in the proper format
+    return {tag["title"]: tag for tag in sorted(data, key=lambda tag: tag["title"].lower())}
```

Comment:
Use `str.casefold()` instead of `.lower()` for key functions.

```suggestion
    return {tag["title"]: tag for tag in sorted(data, key=lambda tag: tag["title"].casefold())}
```

Reply:
This may cause one of the 200 dictator linters to throw up due to line length.

Reply:
426fc71b